### ISSUE TITLE
Hibernate/Spring (and probably other cllients libs and ORMs) cannot take advantage of global routing

### ISSUE THREAD

**Issue by maxenglander** [vitessio/vitess#11940]
# TL;DR

 * Global routing is a nice feature to give PlanetScale enterprise customers a seamless way to use tables across multiple keyspaces.
 * Tools like Hibernate/Spring require you to set a schema (= keyspace), whereas global routing requires that one is not specified.

# Background

For enterprise PlanetScale customers, one of the selling points of PlanetScale (thanks to ✨ Vitess ✨) is the ability to shard their dataset. For customers that are migrating their data into PlanetScale from another service provider, this means that they will likely end up with multiple keyspaces.

In practice this means that their production PlanetScale environment will have multiple keyspaces, whereas their local/dev and non-PS production source environment will be single keyspace. The question we then need to answer is: how do we instruct their application to use multiple keyspaces in production?

There are a couple answers we can give:

 * Use special config/code in production only.
 * Use [global routing](https://vitess.io/docs/15.0/reference/features/global-routing/), and ensure that tables are uniquely named across keyspaces.

For some customers, global routing is the ideal solution. They would prefer not to manage multiple sets of configs for different environments. When the primary customer is a DBA team at a large company, they may have pretty limited ability to change application code or config. 

It is also a nice selling point to be able to tell a customer that to migrate to PlanetScale they just need to repoint their application with a new set of credentials.

# Limitations

Unfortunately for customers using the popular Hibernate/Spring, those tools seem to require that a schema (= keyspace) is explicitly set on the connection, whereas global routing requires that one is not specified.

Telling a customer to interweave the `use @primary` trick mentioned [here](https://github.com/vitessio/vitess/pull/11938#issuecomment-1345624670) would kind of defeat the goal of giving them a seamless migration path into PlanetScale.

# Importance 

In terms of importance of this feature to PS... It's a nice-to-have at this point. We can tell these types of customers that they simply need to add some production-specific configs in order to be able to migrate to use sharding at PlanetScale and it's probably not going to be a deal-breaker, but it would be a nice selling point if we had a more seamless migration path for them :)

**Comment by harshit-gangal** [comment]
With https://github.com/vitessio/vitess/pull/11938 how do you solve the problem with `ddl-auto=validate` as this will result in partial tables only when query `show full tables from <dbname> like '%'` is executed.

**Comment by maxenglander** [comment]
Yes to solve that problem we will need more pieces in place e.g. https://github.com/vitessio/vitess/pull/11882 which I can revive if we decide to go forward with #11938.

**Comment by harshit-gangal** [comment]
We can support this request with a keyspace alias where 2 keyspaces (unsharded and sharded) represent a single keyspace though physically they are 2 different databases. This definition can live inside the VSchema and during planning, the Vitess planner can check the presence of a table in all the alias keyspaces of the given keyspace.
